controllers/storagecluster/storagecluster_controller.go

```diff
@@ -155,6 +156,53 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	)
 
+	enqueueFromStorageProfile := handler.EnqueueRequestsFromMapFunc(
+		func(ctx context.Context, obj client.Object) []reconcile.Request {
+			spobj, ok := obj.(*ocsv1.StorageProfile)
+			if !ok {
+				return []reconcile.Request{}
+			}
+
+			// todo change according to standards
+			if spobj.Status.Phase != util.PhaseReady {
+				return []reconcile.Request{}
+			}
+
+			// Get the StorageCluster objects
+			scList := &ocsv1.StorageClusterList{}
+			err := r.List(r.ctx, scList, &client.ListOptions{Namespace: obj.GetNamespace()})
+			if err != nil {
+				r.Log.Error(err, "Unable to list StorageCluster objects")
+				return []reconcile.Request{}
+			}
+
+			// Return name and namespace per StorageProfile object
+			request := []reconcile.Request{}
+			for _, sc := range scList.Items {
+				// 1. Get the StorageProfile objects in the StorageCluster's ns
+				spList := &ocsv1.StorageProfileList{}
+				err := r.Client.List(r.ctx, spList, &client.ListOptions{Namespace: sc.GetNamespace()})
+				if err != nil {
+					r.Log.Error(err, "Unable to list StorageProfile objects")
+					return []reconcile.Request{}
+				}
+				// 2. For each StorageProfile object push request uniquely identified by the StorageProfile's
+				// ns and StorageCluster's name
+				for _, sp := range spList.Items {
+					request = append(request, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Namespace: sp.Namespace,
+							Name:      sc.Name,
+						},
+					})
+				}
+
+			}
+
+			return request
```

Comment:
Why do we need a for loop if we know that we are only listing a single item? (the list op should contain a limitation for a single result). Also, why are we listing storage profiles? the event already is for a specific storage profile. 

In addition, this code will re-queue multiple SC events for a single SP event. That just does not make sense. The mapping should be 1-1:

```suggestion
			return []reconcile.Request {
				reconcile.Request {
						NamespacedName: sc.NamespacedName
				},
			}
```

Reply:
In that case what differentiates a plain `StorageCluster` event from a `StorageProfile` triggered one? Seems to me as though your suggestion implies we're allowing duplicated events (same identifier) in the queue.

Reply:
Reverted `Items` field since to my understanding `List()` successfully retrieves the objects only when provided with plain arrays, so the for loops are back.
```
type StorageProfileList struct {
	metav1.TypeMeta `json:",inline"`
	metav1.ListMeta `json:"metadata,omitempty"`
	Items           []StorageProfile `json:"items"`
}
```

Reply:
Changed